src/yuzu/configuration/config.cpp

```diff
@@ -773,8 +799,15 @@ void Config::SaveValues() {
     WriteSetting("romsPath", UISettings::values.roms_path);
     WriteSetting("symbolsPath", UISettings::values.symbols_path);
     WriteSetting("screenshotPath", UISettings::values.screenshot_path);
-    WriteSetting("gameListRootDir", UISettings::values.game_directory_path, ".");
-    WriteSetting("gameListDeepScan", UISettings::values.game_directory_deepscan, false);
+    qt_config->beginWriteArray("gamedirs");
+    for (int i = 0; i < UISettings::values.game_dirs.size(); ++i) {
+        qt_config->setArrayIndex(i);
+        const auto& game_dir = UISettings::values.game_dirs.at(i);
```

Comment:
```suggestion
        const auto& game_dir = UISettings::values.game_dirs[i];
```

There's no need to bounds check here when the range of the loop will always be the exact size of the container.